### PR TITLE
MAINT-28561: Avoid not found real users from joining a space

### PR DESCRIPTION
**Pull request by azayati**
This PR will avoid not real users to join spaces, a sanitized url is generated in that case

### CODE REVIEW

**Review by ahamdi — CHANGES_REQUESTED** [review]
This will block the user from joining the space and won't give him a replacement for that. Could we have a sanitized link to let user join the space ?
Could we add a '/' at the end to make sure that the PathParam takes the correct value from URL and change accordingly in the web notification ?

File: component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java [hunk 1]
```diff
@@ -255,11 +258,11 @@ public Response validateRequestToJoinSpace(@PathParam("spaceId") String spaceId,
     checkAuthenticatedRequest();
 
     Space space = getSpaceService().getSpaceById(spaceId);
-    
-    if (space == null) {
+    OrganizationService organizationService = (OrganizationService) getPortalContainer().getComponentInstanceOfType(OrganizationService.class);
+    User user = organizationService.getUserHandler().findUserByName(userId);
```
Comment (ahamdi) [inline component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: You could use identityManager as it performs better that OrganizationService org.exoplatform.social.core.manager.IdentityManagerImpl#getIdentity(java.lang.String, java.lang.String, boolean)

Comment (azayati) [reply component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: done

**Comment by azayati** [comment]
@ahamdi:
> This will block the user from joining the space and won't give him a replacement for that. Could we have a sanitized link to let user join the space ?

Sanitization done for urls containing jsessionId pattern

> Could we add a '/' at the end to make sure that the PathParam takes the correct value from URL and change accordingly in the web notification ?

IMO this is not need for 6.1.x/6.0.x since IntranetNotificationRestService API is used for notification and the concerned API NotificationsRestService is no more used on space members invitation validation. We should check if it is needed and still used on 5.3.x

**Review by ahamdi — APPROVED** [review]

**Pull request merged**